Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,146 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveShared
+import Data
+import Shared
+import WebKit
+
+private let log = Logger.browserLogger
+
+let popup = PaymentHandlerPopupView(imageView: nil, title: Strings.paymentRequestTitle, message: "")
+
+class PaymentRequestExtension: NSObject {
+    fileprivate weak var tab: Tab?
+    fileprivate var response = ""
+    fileprivate var token: String
+    
+    init(tab: Tab) {
+        token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+        self.tab = tab
+    }
+}
+
+extension PaymentRequestExtension: TabContentScript {
+    static func name() -> String {
+        return "PaymentRequest"
+    }
+    
+    func scriptMessageHandlerName() -> String? {
+        return "PaymentRequest"
+    }
+    
+    func sendPaymentRequestError(errorName: String, errorMessage: String) {
+        ensureMainThread {
+            self.tab?.webView?.evaluateJavaScript("PaymentRequestCallback\(self.token).paymentreq_postCreate('', '\(errorName)', '\(errorMessage)')", completionHandler: { _, error in
+                    if error != nil {
+                        log.error(error)
+                    }
+                })
+        }
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        if message.name == "PaymentRequest", let body = message.body as? NSDictionary {
+            do {
+                let messageData = try JSONSerialization.data(withJSONObject: body, options: [])
+                let body = try JSONDecoder().decode(PaymentRequestBodyParser.self, from: messageData)
+                if body.name == "payment-request-show" {
```

Comment:
Same thing, add early return:
```suggestion
                if body.name != "payment-request-show" { return }
```